packages/ra-core/src/reducer/admin/references/oneToMany.ts

```diff
@@ -114,6 +129,46 @@ export const getReferencesByIds = (
     return Object.keys(references).length > 0 ? references : null;
 };
 
+const getRelatedReferences = (previousState, resource) =>
+    Object.keys(previousState).filter(key => key.includes(resource));
+
+const removeDeletedReference = (previousState, action: ActionTypes) => {
+    const relatedTo = getRelatedReferences(previousState, action.meta.resource);
```

Comment:
see the ids reducer: we test that the id exist before making the change. Nothing forbids a component from removing an element which isn't in the store. That would lead to a wrong total.

Reply:
Good catch. I did it for the many case but not here...